### PR TITLE
add Merge Module to the constructor to support multi-layer BiRNN

### PR DESCRIPTION
**Pull request by JerryYanWan**
## What changes were proposed in this pull request?
add Merge Module to the constructor to support multi-layer BiRNN

## How was this patch tested?
Unit Test
Test the correctness by comparing it to Element research's brnn

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala [hunk 1]
```diff
@@ -20,11 +20,12 @@ import com.intel.analytics.bigdl._
...
 import com.intel.analytics.bigdl.tensor.Tensor
 import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
-import com.intel.analytics.bigdl.utils.T
+import com.intel.analytics.bigdl.utils.{T, Table}
 
 import scala.reflect.ClassTag
 
-class BiRecurrent[T : ClassTag] ()
+class BiRecurrent[T : ClassTag] (
+  merge: AbstractModule[Table, Tensor[T], T])
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala]: add a default value?

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala [hunk 2]
```diff
@@ -115,19 +116,23 @@ class BiRecurrent[T : ClassTag] ()
...
-    val state = Seq(super.hashCode(), birnn)
+    val state = Seq(super.hashCode(), timeDim, layer, revLayer, birnn)
     state.map(_.hashCode()).foldLeft(0)((a, b) => 31 * a + b)
   }
 }
 
 object BiRecurrent {
-  def apply[@specialized(Float, Double) T: ClassTag]()
+  def apply[@specialized(Float, Double) T: ClassTag](
+    merge: AbstractModule[Table, Tensor[T], T])
```
Comment (jason-dai) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala]: add a default value such as null?

Comment (JerryYanWan) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/BiRecurrent.scala]: added, if merge is null, model.add(CAddTable)

**Comment by jason-dai** [comment]
LGTM

**Pull request merged**